### PR TITLE
returning content-type application/json on validation error

### LINKED ISSUE #720: Validation message are now `text/plain`
Validation messages so far have been of type `Content-Type: application/json`, resembling something like this: 

```bash
$ http post api.mydomain.com/hello
HTTP/1.1 400 Bad Request
connection: close
content-length: 59
content-type: text/plain
date: Thu, 08 Sep 2022 07:48:54 GMT
server: envoy

{
	"message": "Request body has an error: value is required but missing"
}
```

Now it looks like this:

```bash
$ http post api.mydomain.com/hello
HTTP/1.1 400 Bad Request
connection: close
content-length: 59
content-type: text/plain
date: Thu, 08 Sep 2022 07:48:54 GMT
server: envoy

request body has an error: value is required but missing |
```

Which even broke the more detailed error message: 
```bash
$ http post api.mydomain.com/hello namee="abdallah"
HTTP/1.1 400 Bad Request
connection: close
content-length: 295
content-type: text/plain
date: Thu, 08 Sep 2022 07:47:47 GMT
server: envoy

request body has an error: doesn't match the schema: Error at "/name": property "name" is missing
Schema:
  {
    "properties": {
      "name": {
        "type": "string"
      }
    },
    "required": [
      "name"
    ],
    "type": "object"
  }

Value:
  {
    "namee": "abdallah"
  }
 |  |
```

### PR DESCRIPTION
**Pull request by jasmingacic**
Signed-off-by: jasmingacic <jasmin.gacic@gmail.com>
Fixes #720

### CODE REVIEW

File: internal/validation/extension.go [hunk 1]
```diff
@@ -122,14 +123,28 @@ func (s *Server) Process(srv pb.ExternalProcessor_ProcessServer) error {
 				}
 				err = s.validate(req, service, operation)
 				if err != nil {
+					errMsg := ErrorBody{}
```
Comment (mbana) [inline internal/validation/extension.go:126]: I suspect this is intentional but why not use and define `NewErrorBody` then continue to use `errMsg.Error`?

File: internal/validation/extension.go [hunk 2]
```diff
@@ -122,14 +123,28 @@ func (s *Server) Process(srv pb.ExternalProcessor_ProcessServer) error {
...
 						Response: &pb.ProcessingResponse_ImmediateResponse{
 							ImmediateResponse: &pb.ImmediateResponse{
 								Status: &v32.HttpStatus{Code: v32.StatusCode_BadRequest},
-								Body:   err.Error(),
+								Body:   errMsg.Error,
+								Headers: &pb.HeaderMutation{
+									SetHeaders: []*v31.HeaderValueOption{
+										{
+											Header: &v31.HeaderValue{
+												Key:   "Content-Type",
```
Comment (mbana) [inline internal/validation/extension.go:138]: Make this a constant please, it is repeated elsewhere.

File: internal/validation/extension.go [hunk 3]
```diff
@@ -122,14 +123,28 @@ func (s *Server) Process(srv pb.ExternalProcessor_ProcessServer) error {
...
+												Value: "applictaion/json",
```
Comment (mbana) [inline internal/validation/extension.go:139]: Make this a constant please, it is repeated elsewhere.

File: internal/validation/extension.go [hunk 4]
```diff
@@ -175,11 +190,24 @@ func (s *Server) Process(srv pb.ExternalProcessor_ProcessServer) error {
...
 							ImmediateResponse: &pb.ImmediateResponse{
 								Status: &v32.HttpStatus{Code: v32.StatusCode_BadRequest},
-								Body:   err.Error(),
+								Body:   errorMsg.Error,
+								Headers: &pb.HeaderMutation{
+									SetHeaders: []*v31.HeaderValueOption{
+										{
+											Header: &v31.HeaderValue{
+												Key:   "Content-Type",
+												Value: "applictaion/json",
```
Comment (mbana) [inline internal/validation/extension.go:206]: Make this a constant please, it is repeated elsewhere.

File: internal/validation/extension.go [hunk 5]
```diff
@@ -175,11 +190,24 @@ func (s *Server) Process(srv pb.ExternalProcessor_ProcessServer) error {
...
 						Response: &pb.ProcessingResponse_ImmediateResponse{
 							ImmediateResponse: &pb.ImmediateResponse{
 								Status: &v32.HttpStatus{Code: v32.StatusCode_BadRequest},
-								Body:   err.Error(),
+								Body:   errorMsg.Error,
+								Headers: &pb.HeaderMutation{
+									SetHeaders: []*v31.HeaderValueOption{
+										{
+											Header: &v31.HeaderValue{
+												Key:   "Content-Type",
```
Comment (mbana) [inline internal/validation/extension.go:205]: Make this a constant please, it is repeated elsewhere.

File: internal/validation/extension.go [hunk 6]
```diff
@@ -238,3 +266,15 @@ func (s *Server) validate(r *http.Request, service *Service, operation *operatio
...
 	})
 }
+
+type ErrorBody struct {
+	Error string `json:"error,omitempty"`
+}
+
+func (e *ErrorBody) SetErrorBody(err error) {
+	errorMsg := ErrorBody{Error: err.Error()}
+	jsn, _ := json.Marshal(errorMsg)
```
Comment (mbana) [inline internal/validation/extension.go:276]: Are you intentionally ignoring the `error` from here?

**Review by mbana — CHANGES_REQUESTED** [review]
Few comments about things I am not sure about.

File: internal/validation/extension.go (hunk 6 above)
Comment (jasmingacic) [reply internal/validation/extension.go:276]: yes, as  there is no point in doing it. Either we get the error message or we don't

File: internal/validation/extension.go (hunk 1 above)
Comment (jasmingacic) [reply internal/validation/extension.go]: Thanks, i got sidetracked with something else

File: internal/validation/extension.go (hunk 5 above)
Comment (jasmingacic) [reply internal/validation/extension.go]: done

File: internal/validation/extension.go (hunk 4 above)
Comment (jasmingacic) [reply internal/validation/extension.go]: done

File: internal/validation/extension.go (hunk 2 above)
Comment (jasmingacic) [reply internal/validation/extension.go]: done

**Review by mbana — APPROVED** [review]
LGTM.

**Pull request merged**